### PR TITLE
FT: AZURE GET

### PR DESCRIPTION
**Pull request by nicolas2bert**
FT: AZURE GET

### CODE REVIEW

File: lib/api/objectGet.js [hunk 1]
```diff
@@ -106,6 +111,11 @@ function objectGet(authInfo, request, returnTagCount, log, callback) {
             // objMD.location is just a string
             dataLocator = Array.isArray(objMD.location) ?
                 objMD.location : [{ key: objMD.location }];
+
+            if (dataLocator && dataLocator[0] &&
```
Comment (LaurenSpiegel) [inline lib/api/objectGet.js]: add comment:
// if the data backend is azure, there will only ever be at most one item in the dataLocator array

File: tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            .catch(err => {
+                process.stdout.write('Error emptying/deleting bucket: ' +
+                `${err}\n`);
+                throw err;
+            });
+        });
+        keys.forEach(key => {
+            describe(`${key.describe} size`, () => {
+                before(done => {
+                    s3.putObjectAsync({
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js]: why Async?

File: tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                    assert.strictEqual(res.ContentRange, 'bytes 4-9/10');
+                    assert.strictEqual(res.Body.toString(), '456789');
+                    done();
+                });
+            });
+        });
+
+        describe('returning error', () => {
+            before(done => {
+                s3.putObjectAsync({
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js]: same

File: tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                    err => {
+                        assert.equal(err, null, 'Expected success but got ' +
+                        `error ${err}`);
+                        done(err);
+                    });
+                });
+            });
+
+            it('should return an error on get done to object deleted ' +
+            'from AWS', done => {
```
Comment (LaurenSpiegel) [inline tests/functional/aws-node-sdk/test/multipleBackend/get/getAzure.js]: azure

File: lib/data/external/AzureClient.js [hunk 5]
```diff
@@ -39,14 +39,34 @@ class AzureClient {
...
+        // for backwards compatibility
+        const key = typeof(objectGetInfo) === 'string' ? objectGetInfo :
+          objectGetInfo.key;
+        const response = objectGetInfo.response;
+        const azureStreamingOptions = objectGetInfo.azureStreamingOptions;
+        this._client.getBlobToStream(this._azureContainerName, key, response,
+          azureStreamingOptions, err => {
+              if (err) {
+                  const log = createLogger(reqUids);
+                  log.error('err from Azure GET data backend',
```
Comment (LaurenSpiegel) [inline lib/data/external/AzureClient.js]: does this actually log an error "not found" on your error test? Even if it does, we should consider doing a head call to azure first to see if we are going to get back an error so we can fail more gracefully.

File: tests/functional/aws-node-sdk/test/multipleBackend/utils.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    if (config.locationConstraints[azureLocation] &&
+    config.locationConstraints[azureLocation].details &&
+    config.locationConstraints[azureLocation].details.azureContainerName) {
+        azureContainerName =
+          config.locationConstraints[azureLocation].details.azureContainerName;
+    }
+    return azureContainerName;
+};
+
+utils.getKeysForTestingAzure = () => {
```
Comment (dora-korpar) [inline tests/functional/aws-node-sdk/test/multipleBackend/utils.js]: I'd rename this to 'getAzureKeys' for consistency

**Pull request merged**